ffmpeg/filter.c

```diff
@@ -204,14 +225,19 @@ int init_audio_filters(struct input_ctx *ictx, struct output_ctx *octx)
 int filtergraph_write(AVFrame *inf, struct input_ctx *ictx, struct output_ctx *octx, struct filter_ctx *filter, int is_video)
 {
   int ret = 0;
-  // Sometimes we have to reset the filter if the HW context is updated
-  // because we initially set the filter before the decoder is fully ready
-  // and the decoder may change HW params
+  // We have to reset the filter because we initially set the filter
+  // before the decoder is fully ready, and the decoder may change HW params
+  // XXX: Unclear if this path is hit on all devices
   if (is_video && inf && inf->hw_frames_ctx && filter->hwframes &&
       inf->hw_frames_ctx->data != filter->hwframes) {
     free_filter(&octx->vf); // XXX really should flush filter first
+    if (octx->dnn_filtergraph) {
+      // swap filtergraph with the pre-initialized DNN filtergraph
+      octx->vf.graph = *octx->dnn_filtergraph;
+    }
     ret = init_video_filters(ictx, octx);
     if (ret < 0) return lpms_ERR_FILTERS;
+    struct filter_ctx *vf = filter;
```

Comment:
What does this line do? Doesn't seem like the var is used elsewhere?

Reply:
Ah I have no idea when I wrote that, will remove!